lib/core.js

```diff
@@ -1338,9 +1338,14 @@
 
 
         options.getRequestOptions = function(path, defaultValue) {
-                         
+
             var alternatePath = path.replace(/[^.]+\./, '_.');
-            return options.getProviderOptions(alternatePath, options.getProviderOptions(path, defaultValue));
+            var value = options.getProviderOptions(alternatePath, options.getProviderOptions(path, defaultValue));
+            if (typeof value !== typeof defaultValue) {
+                // Type mismatch fallback.
+                return defaultValue;
```

Comment:
`defaultValue` can be undefined, so we can do fallback only if its defined.

Example for test:
https://github.com/itteco/iframely/blob/master/plugins/domains/codepen.io.js#L45